kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/modem/quectel/eg25/QuectelEG25.java

```
@@ -0,0 +1,336 @@
+            if (reply != null) {
+                String sCgclass = this.getResponseString(reply);
+                if (sCgclass.startsWith("+CGCLASS:")) {
+                    sCgclass = sCgclass.substring("+CGCLASS:".length()).trim();
+                    if (sCgclass.equals("\"A\"")) {
+                        serviceType = "UMTS";
+                    } else if (sCgclass.equals("\"B\"")) {
+                        serviceType = "GSM/GPRS";
+                    } else if (sCgclass.equals("\"CG\"")) {
+                        serviceType = "GPRS";
+                    } else if (sCgclass.equals("\"CC\"")) {
+                        serviceType = "GSM";
+                    }
+                }
+            }
+        }
+
+        return serviceType;
+    }
+
+    @Override
+    public List<ModemTechnologyType> getTechnologyTypes() throws KuraException {
+
+        List<ModemTechnologyType> modemTechnologyTypes = null;
+        ModemDevice device = getModemDevice();
+        if (device == null) {
+            throw new KuraException(KuraErrorCode.UNAVAILABLE_DEVICE, "No modem device");
+        }
+        if (device instanceof UsbModemDevice) {
+            SupportedUsbModemInfo usbModemInfo = SupportedUsbModemsInfo.getModem((UsbModemDevice) device);
+            if (usbModemInfo != null) {
+                modemTechnologyTypes = usbModemInfo.getTechnologyTypes();
+            } else {
+                throw new KuraException(KuraErrorCode.UNAVAILABLE_DEVICE, "No usbModemInfo available");
+            }
+        } else {
+            throw new KuraException(KuraErrorCode.UNAVAILABLE_DEVICE, "Unsupported modem device");
+        }
+        return modemTechnologyTypes;
+    }
+
+    @Override
+    public boolean isGpsSupported() throws KuraException {
+        return false; // Will be activated later
+    }
+
+    @Override
+    public void enableGps() throws KuraException {
+        logger.warn("Modem GPS not supported");
```

Maybe here we can throw an UnsupportedOperationException?